esp-hal/src/lib.rs

```diff
@@ -582,16 +582,7 @@ impl LockState {
     pub const fn new() -> Self {
         Self {
             #[cfg(multi_core)]
-            core: portable_atomic::AtomicU8::new(0),
-        }
-    }
-
-    #[cfg(multi_core)]
-    fn current_core() -> u8 {
-        // Note: 0 means unlocked.
-        match get_core() {
-            Cpu::ProCpu => 1,
-            Cpu::AppCpu => 2,
+            core: portable_atomic::AtomicU8::new(0xFF),
```

Comment:
Do you have something against AtomicUsize? Xtensa has L32EX/S32EX, but nothing narrower, for example, and I'm not sure what portable-atomic does here.

Reply:
Ohhh, it's a good thing you said. I honestly wasn't sure which primitive type would be most optimal for all multicore chips. I can use usize sure.

> I'm not sure what portable-atomic does here.

I'll happily remove it. I was worried I'd be told off for not using it :sweat_smile: .

Reply:
> I'll happily remove it. I was worried I'd be told off for not using it.

No sorry I meant I'm not sure if p-a just widens AtomicU8 to U32/usize anyway, or what. Definitely keep using portable-atomic, please.

Reply:
Ah crap, too late. I figure it'd widen? My thought process was "Chose the smallest type for my requirements and the library/compiler would pick something sensible"
I'll bring back portable-atomic.